src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -699,17 +699,41 @@ static ucs_status_t uct_rc_mlx5_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
-                                   const uct_device_addr_t *dev_addr,
-                                   const uct_iface_addr_t *iface_addr)
+int uct_rc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                      const uct_iface_is_reachable_params_t *params)
 {
-    uint8_t my_type = uct_rc_mlx5_iface_get_address_type(tl_iface);
+    const uct_iface_addr_t *iface_addr = params->iface_addr;
+    char* info_string                  = params->info_string;
+    size_t info_string_length          = params->info_string_length;
+    uct_ib_iface_t *iface              = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    uint8_t my_type                    = uct_rc_mlx5_iface_get_address_type(tl_iface);
 
     if ((iface_addr != NULL) && (my_type != *(uint8_t*)iface_addr)) {
+        if (info_string && info_string_length > 0) {
```

Comment:
change to if (info_string !=NULL && info_string_length > 0)
And check all other places.

Reply:
Done